src/server/scripts/EasternKingdoms/ScarletMonastery/instance_scarlet_monastery.cpp

```diff
@@ -262,63 +344,97 @@ class npc_mograine : public CreatureScript
-                    mograine->DespawnOrUnsummon(3 * IN_MILLISECONDS);
-                    mograine->Kill(me, me, true);
-                    return 0;
-                default:
-                    if(mograine)
-                        mograine->DespawnOrUnsummon(0);
-                    return 0;
+            case EVENT_MOGRAINE_FACING_PLAYER:
+                me->SetFacingToObject(playerWhoStartedAshbringer);
+                events.ScheduleEvent(EVENT_MOGRAINE_KNEEL, 1s, 3s);
+                break;
+            case EVENT_MOGRAINE_KNEEL:
+                me->SetSheath(SHEATH_STATE_UNARMED);
+                me->SetStandState(UNIT_STAND_STATE_KNEEL);
+                events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK3, 1s, 2s);
+                break;
+            case EVENT_MOGRAINE_EMOTE_TALK3:
+                me->HandleEmoteCommand(EMOTE_ONESHOT_TALK);
+                me->AI()->Talk(SAY_MO_AB_TALK3, playerWhoStartedAshbringer);
+                break;
+            case EVENT_SUMMONED_HIGHLORD_MOGRAINE:
+                if (Creature* summonedMograine = me->SummonCreature(NPC_HIGHLORD_MOGRAINE, 1033.4642f, 1399.1022f, 27.337427f, 6.257956981658935546f, TEMPSUMMON_TIMED_DESPAWN, 120000))
+                {
+                    summonedMograine->SetFaction(FACTION_FRIENDLY);
+                    summonedMograine->SetUInt32Value(UNIT_VIRTUAL_ITEM_SLOT_ID + 0, EQUIP_UNEQUIP);
+                    summonedMograine->SetSpeed(MOVE_WALK, 1.0f);
+                    summonedMograine->SetDisplayId(MODEL_HIGHLORD_MOGRAINE);
+                    summonedMograine->setActive(true);
+                    summonedMograine->CastSpell(summonedMograine, SPELL_MOGRAINE_COMETH_DND, false);//Sniffing data shows the use of this spell transformation, but the dispersion effect of this spell is not seen in the video
+                }
+                events.ScheduleEvent(EVENT_HIGHLORD_MOGRAINE_MOVE_STOP, 48500ms);
+                break;
+            case EVENT_HIGHLORD_MOGRAINE_MOVE_STOP:
+                summonedMograine->StopMovingOnCurrentPos();
+                summonedMograine->HandleEmoteCommand(EMOTE_ONESHOT_POINT);
+                summonedMograine->AI()->Talk(SAY_HM_AB_TALK0, 200ms);
+                events.ScheduleEvent(EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE, 3000ms);
+                break;
+            case EVENT_MOGRAINE_FACING_HIGHLORD_MOGRAINE:
+                me->SetFacingToObject(summonedMograine);
+                events.ScheduleEvent(EVENT_MOGRAINE_UNIT_STAND_STATE_STAND, 400ms);
+                break;
+            case EVENT_MOGRAINE_UNIT_STAND_STATE_STAND:
+                me->SetStandState(UNIT_STAND_STATE_STAND);
+                me->SetSheath(SHEATH_STATE_MELEE);
+                events.ScheduleEvent(EVENT_MOGRAINE_EMOTE_TALK4, 800ms);
+                break;
+            case EVENT_MOGRAINE_EMOTE_TALK4:
+                me->HandleEmoteCommand(EMOTE_ONESHOT_QUESTION);
+                me->AI()->Talk(SAY_MO_AB_TALK4, 200ms);
```

Comment:
same goes for all these and below

Reply:
Here there is a 200+ ms delay in the expression in sniffing. More than 1-99ms above mentioned above